stdlib/Logging/docs/src/index.md

```diff
@@ -0,0 +1,202 @@
+  - Use `Warn` when a potential problem is detected.
+  - Use `Error` to report errors where the code has enough context to recover
+    and continue.  (When the code doesn't have enough context, an exception or
+    early return is more appropriate.)
+* The *message* is a human readable string describing the event in markdown
+  format.
+* Optional *key value pairs* allow arbitrary data to be attached to each event.
+  Some keys have conventional meaning which can affect the way an event is
+  interpreted (see [`@logmsg`](@ref)).
+
+The system also generates some standard information for each event:
+
+* The `module` in which the logging macro was expanded.
+* The `file` and `line` where the logging macro occurs in the source code.
+* A message `id` which is unique for each logging macro invocation. This is
+  very useful as a key for caching information or actions associated with an
+  event. For instance, it can be used to limit the number of times a message
+  is presented to the user.
+* A `group` for the event which is set to the base name of the file by default,
+  without extension.  This can be used to group messages into categories more
+  finely than the log level (for example, all deprecation warnings have group
+  `:depwarn`), or into logical groupings across or within modules.
+
+## Processing log events
+
+As you can see in the examples, logging statements make no mention of
+where log events go or how they are processed. This is a key design feature
+which makes the system composable and natural for concurrent use. It does this
+by separating two different concerns:
+
+* *Creating* log events is the concern of the module author who needs to
+  decide where events are triggered and which information to include.
+* *Processing* of log events - that is, display, filtering, aggregation and
+  recording - is the concern of the application author who needs to bring
+  multiple modules together into a cooperating application.
+
+### Loggers
+
+Processing of events is performed by a *logger* which is the first piece of
+user configurable code to see the event. All loggers must be subtypes of
+[`AbstractLogger`](@ref).
+
+When an event is triggered, the appropriate logger is found by looking for a
+task local logger with the global logger as fallback.  The idea here is that
+the application code knows how log events should be processed and exists
+somewhere at the top of the call stack. So we should look up through the call
+stack to discover the logger - that is, the logger should be *dynamically
+scoped*. (This is a point of contrast with logging frameworks where the
+logger is *lexically scoped* - provided explicitly by the module author or a
+simple global variable. In such a system it's awkward to control logging while
+composing functionality from multiple modules.)
+
+The global logger may be set with [`global_logger`](@ref), and task local
+loggers controlled using [`with_logger`](@ref).  Newly spawned tasks inherit
```

Comment:
Newly-spawned

Reply:
Thanks, any gifts of compound adjective detection are gratefully received :-)

Reply:
task-local in the line above too!

Reply:
Actually, this is incorrect: you *don't* hyphenate compound adjectives where the modifier is an adverb ending with "ly":  https://www.merriam-webster.com/words-at-play/6-common-hypercorrections-and-how-to-avoid-them/hyphenating-ly-adverbs

Reply:
^ that's true, my bad!